apps/monorail/views/nodes/NodesGrid.js

```diff
@@ -60,7 +74,18 @@ export default class NodesGrid extends Component {
 
   listNodes() {
     this.setState({loading: true});
-    this.nodes.list().then(() => this.setState({loading: false}));
+    this.nodes.list().then(() => {
+      let promises = [];
+      this.nodes.each(node => {
+        let promise = new Promise((resolve) => {
+          this.catalogs.relateNode(node, 'dmi', this.nodes).then(resolve, resolve);
```

Comment:
It works even if there is no DMI. The Enclosure Node 0 in my screenshot doesn't have a DMI catalog.